inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -941,6 +941,39 @@ def normalize_collaborations(obj, eng):
     return obj
 
 
+def _get_correct_affiliation_data(affiliations):
+    cleaned_affiliations = []
+    for aff in affiliations:
+        cleaned_affiliations.append({key: val for key, val in aff.items() if key in ['value', 'record']})
+    return cleaned_affiliations
+
+
+def _extract_matched_aff_from_highlight(highlighted_raw_affs, author_raw_affs, author_affs):
+    highlight_matches = [re.finditer('(?=<em>).*?(?<=<\/em>)', raw_aff) for raw_aff in highlighted_raw_affs]
+    raw_aff_highlight_lenghts = [sum([match.span()[1] - match.span()[0] for match in aff_matches]) for aff_matches in highlight_matches]
+    longest_highlight_idx = raw_aff_highlight_lenghts.index(max(raw_aff_highlight_lenghts))
+    extracted_raw_aff = re.sub('<em>|</em>', "", highlighted_raw_affs[longest_highlight_idx])
+    for ix, raw_aff in enumerate(author_raw_affs):
+        if raw_aff['value'] == extracted_raw_aff:
+            return [author_affs[ix]]
```

Comment:
if you want to iterate on two lists together, you should use `zip`, not `enumerate`:
```suggestion
    for raw_aff, aff in zip(author_raw_affs, author_affs):
        if raw_aff['value'] == extracted_raw_aff:
            return aff
```